CS/vehicalDekho/xmlParser/DataXmlParcer.java

```diff
@@ -0,0 +1,196 @@
+package xmlParser;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.events.Attribute;
+import javax.xml.stream.events.EndElement;
+import javax.xml.stream.events.StartElement;
+import javax.xml.stream.events.XMLEvent;
+
+import model.Bike;
+import model.Car;
+import model.Vehicle;
+
+public class DataXmlParcer {
+	static private String ID = "id";
+	static private String MAKE = "make";
+	static private String MODEL = "model";
+	static private String ENGININCC = "enginInCC";
+	static private String FUELCAPACITY = "fuelCapacity";
+	static private String MILAGE = "milage";
+	static private String PRICE = "price";
+	static private String ROADTAX = "roadTax";
+	static private String SELFSTART = "selfStart";
+	static private String HELMATPRIZE = "helmetPrize";
+	static private String AC = "AC";
+	static private String POWERSTEERING = "powerSteering";
+	static private String ACCESSORYkIT = "accessoryKit";
+
+	@SuppressWarnings({ "unchecked", "null" })
+	public static List<Vehicle> readData(String configFile) {
+		List<Vehicle> listOfVehicles = new ArrayList<Vehicle>();
+		try {
+
+			XMLInputFactory inputFactory = XMLInputFactory.newInstance();
+
+			InputStream in = new FileInputStream(configFile);
+			XMLEventReader eventReader = inputFactory.createXMLEventReader(in);
+
+			Vehicle vehicle = null;
+
+			while (eventReader.hasNext()) {
+				XMLEvent event = eventReader.nextEvent();
+
+				if (event.isStartElement()) {
+					StartElement startElement = event.asStartElement();
+
+					if (startElement.getName().getLocalPart() == ("vehicle")) {
+
+						Iterator<Attribute> attributes = startElement
+								.getAttributes();
+						while (attributes.hasNext()) {
+							Attribute attribute = attributes.next();
+							if (attribute.getName().toString().equals("type")) {
+								if (attribute.getValue().equals("car")) {
+									vehicle = new Car();
+
+								} else {
```

Comment:
ACTION: better use else-if in such case, because the attribute might not be only "bike" if it is not "car". In such a case you should throw exception.